pydash/strings.py

```diff
@@ -736,6 +736,56 @@ def lines(text):
     return text.splitlines()
 
 
+def lower_case(text):
+    """Converts string to lower case as space separated words. Before
+    converting the given text to lower case this method:
+
+      - replaces special character with white spaces,
+      - adds a white space before the capital letter,
+      - replaces multiple whitespaces with a single whitespace,
+      - pertains spaces between words in the given text,
+      - pertains the digits given in the text.
+
```

Comment:
I don't think we need to go into this much detail in describing what the function does. The Lodash description is simply `Converts string, as space separated words, to lower case.` which I think would be fine here. Since we have the `words()` function, you could link to it to give more context for what a "word" is.